confuse.py

```diff
@@ -566,13 +724,17 @@ def __init__(self, sources):
         self.redactions = set()
 
     def add(self, obj):
-        self.sources.append(ConfigSource.of(obj))
+        src = ConfigSource.of(obj)
+        self.sources.append(src)
+        return src
```

Comment:
It looks like `add` and `set` (below) both now return the newly-created configuration source. That seems cool enough, but can you perhaps elaborate a bit on the use case you had in mind? In a way it doesn't seem _that_ hard for clients to keep a reference to their own source that they pass in, if that's wha they want. (Or maybe this is used somewhere else internally?) In any case, it would be great to add docs for this new behavior too if we need it…

Reply:
This was more for sources that are implicitly created internally using ConfigSource.of:
```
config.add({'a': 5})
```
It's not essential, but it makes it less error prone than using indexes. I could be talked into removing this tho